adapters/repos/db/shard.go

```diff
@@ -227,14 +227,21 @@ func NewShard(ctx context.Context, promMetrics *monitoring.PrometheusMetrics,
 		return nil, err
 	}
 
+	s.NotifyReady()
+
 	s.queue, err = NewIndexQueue(s.ID(), s, s.VectorIndex(), s.centralJobQueue, s.indexCheckpoints, IndexQueueOptions{Logger: s.index.logger})
 	if err != nil {
 		return nil, err
 	}
 
-	err = s.queue.PreloadShard(ctx, s)
-	if err != nil {
-		return nil, err
+	if asyncEnabled() {
+		go func() {
```

Comment:
IMO routine should be stopped if shard will receive either `shutdown` or `drop` request, in case it runs longer that shards live

Reply:
It's stopped automatically when the queue is closed actually, it's just that the logic is in the method.